tests/resources/test_resources_user_access.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Tests for RDMGrantsAccessResource."""
+import json
+
+
+def test_read_by_subject_found(
```

Comment:
For next PR: missing permissions checks.
Can you please add 2 simple tests:
- as anonymous, I get/post grants and I get 401
- as another user, I get/post grants and I get 403
any other test?